### PR TITLE
Basic Faithless Mob

### PR DESCRIPTION
**Pull request by Comxy**
## About The Pull Request
Turns the faithless mob into a basic mob with unique behaviors, it will now break lights and drag around victims bodies. Can also punch open airlocks if they can be opened.
## Why It's Good For The Game
Now the faithless mob has its own unique behavior and it is cool to see more AI variety.
## Changelog
:cl:
add: Faithless will now also break lights and drag victims around
refactor: Faithless into basic mob
/:cl:

### CODE REVIEW

File: code/datums/ai/basic_mobs/basic_subtrees/attack_obstacle_in_path.dm [hunk 1]
```diff
@@ -25,6 +25,8 @@
 	action_cooldown = 1 SECONDS
 	/// If we should attack walls, be prepared for complaints about breaches
 	var/can_attack_turfs = FALSE
+	/// Tries to bump open airlocks with an attack
+	var/bump_open_airlock = FALSE
```
Comment (Jacquerel) [inline code/datums/ai/basic_mobs/basic_subtrees/attack_obstacle_in_path.dm:29]: Would this be better as a component which picks up the mob attempting to attack a door and tries to open it instead? That seems like it'd be more exetendable.

**Comment by Shadowflame909** [comment]
The faithless are shoeless glass stunners.

Ash-Lizards watch out!

File: code/datums/ai/basic_mobs/basic_subtrees/attack_obstacle_in_path.dm (hunk 1 above)
Comment (CapybaraExtravagante) [reply code/datums/ai/basic_mobs/basic_subtrees/attack_obstacle_in_path.dm:29]: That probably would be better

**Comment by itseasytosee** [comment]
Do the nanotrasen hostile and retaliate mobs next. They always seemed cool if they weren't so bugged.

**Review by tralezab — CHANGES_REQUESTED** [review]
implement Jacquerel's review idea

File: code/datums/ai/basic_mobs/basic_subtrees/attack_obstacle_in_path.dm (hunk 1 above)
Comment (Comxy) [reply code/datums/ai/basic_mobs/basic_subtrees/attack_obstacle_in_path.dm:29]: Got it

File: code/modules/mob/living/basic/space_fauna/faithless/faithless.dm [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	/// How long do we paralyze a target for if we attack them
+	var/paralyze_duration = 2 SECONDS
+
+/datum/ai_behavior/basic_melee_attack/faithless/perform(delta_time, datum/ai_controller/controller, target_key, targetting_datum_key, hiding_location_key)
+	. = ..()
+	var/datum/weakref/weak_target = controller.blackboard[target_key]
+	var/atom/target = weak_target?.resolve()
+	var/mob/living/living_pawn = controller.pawn
+
+	if(isliving(target))
```
Comment (jlsnow301) [inline code/modules/mob/living/basic/space_fauna/faithless/faithless.dm:74]: can you early return here

Comment (Comxy) [reply code/modules/mob/living/basic/space_fauna/faithless/faithless.dm:74]: Good point

**Pull request merged**